editor/editor_properties.cpp

```diff
@@ -1941,6 +1941,30 @@ void EditorPropertyQuaternion::_set_read_only(bool p_read_only) {
 	}
 }
 
+void EditorPropertyQuaternion::_edit_normalize_quaternion_value() {
+	if (normalize_quaternion_bttn->is_pressed()) {
+		EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+		undo_redo->create_action("Quaternion Normalized");
+
+		Quaternion temp;
+		for (int i = 0; i < 4; i++) {
+			temp[i] = (real_t)spin[i]->get_value();
+
+			//undo_redo->add_undo_property(spin[i], "Un-Normalize Quaternion", spin[i]);
+			undo_redo->add_undo_property(get_edited_object(), get_edited_property(), spin[i]);
+		}
+		temp = temp.normalized();
+		for (int i = 0; i < 4; i++) {
+			spin[i]->set_value_no_signal((double)temp[i]);
+
+			undo_redo->add_undo_property(get_edited_object(), get_edited_property(), temp[i]);
+			//undo_redo->add_do_property(spin[i], "Re-Normalize Quaternion", temp[i]);
+		}
+		_value_changed(-1, "");
+		undo_redo->commit_action();
```

Comment:
If property's `set_value()` remains in the history, then I assume undo/redo may be enabled without `add_undo_property()`. In other words, maybe we should consider a way to avoid the need for the EditorUndoRedoManager here.